### DISCUSSION TITLE
How compare localdatetime

### DISCUSSION THREAD

**Discussion by SShahmuradli** [Blazebit/blaze-persistence#1705] [General]
Hi! i try to select data where localdatetime field(createdOn) match to period between to 2 localdatetimes

what i try to do, but it does not work. 

```
 public List<Somedata> findsomedata() {


      var sixtyDays = LocalDateTime.now().minusDays(60);    
       var query = persistence.createCriteriaBuilder(Somedata.class);
        return query.where(Fields.createdOn)
            .betweenExpression(String.valueOf(sixtyDays)).andExpression(String.valueOf(sixtyDays.minusHours(3)))
            .getResultList();
    } 
    
```

    How can i solve it? Thanks in advance!

**Comment by beikov** [comment]
You can use the following:

```java
return query.where(Fields.createdOn)
        .between(sixtyDays).and(sixtyDays.minusHours(3))
        .getResultList();
```

Note that `betweenExpression` and `andExpression` expect a JPQL expression, but what you want to pass here is an actual value.

**Reply by SShahmuradli** [reply]
<img width="1058" alt="image" src="https://user-images.githubusercontent.com/113012735/223078112-4b2aa2be-9d55-4640-b002-20297aadebab.png">
<img width="541" alt="image" src="https://user-images.githubusercontent.com/113012735/223078517-50174124-5924-49a1-a8bc-d991dd1038ea.png">
you can see that createdOn field value between 2 dates, but i can not get this row from table

**Reply by beikov** [reply]
Can you please post the SQL query that Hibernate logs out and the parameters it sets? Chances are, that this is related to a time zone setting. Are you running your application and database in the UTC time zone?

**Reply by SShahmuradli** [reply]
Hibernate: 
    select
        us0_.name as name1_11_,
        us0_.createdOn as createdOn2_11_ 
    from
        customer.mytable us0_ 
    where
        us0_.createdOn between ? and ?

**Reply by SShahmuradli** [reply]
yes, app and db running in UTC time zone

**Reply by beikov** [reply]
Did you configure the `hibernate.jdbc.time_zone` property? Also, did you configure the `user.timezone=UTC` system property for your JVM? Just trying to make sure, because that kind of is relevant to how the JDBC driver interacts with the database.

**Reply by SShahmuradli** [reply]
anyway i run app and db in my local machine and there should not be problem with timezone, right?

**Reply by SShahmuradli** [reply]
i have it in application.properties quarkus.hibernate-orm.jdbc.timezone=UTC

**Reply by SShahmuradli** [reply]
did not configure anything in JVM

**Reply by beikov** [reply]
>anyway i run app and db in my local machine and there should not be problem with timezone, right?

You have to configure the JVM system property `user.timezone=UTC`, otherwise the JDBC driver might do funky things. Also see https://stackoverflow.com/questions/59896136/how-to-set-timezone-in-quarkus-to-utc

**Reply by SShahmuradli** [reply]
thanks. It was UTC problem. for local testing took into account the time zone)

**Reply by beikov** [reply]
Time zones are hard. The best run into issues with them ;)
